x-pack/plugins/observability_ai_assistant/public/components/message_panel/message_text.tsx

```diff
@@ -86,20 +88,56 @@ const loadingCursorPlugin = () => {
   };
 };
 
-export function MessageText({ loading, content }: Props) {
+const esqlLanguagePlugin = () => {
+  const visitor = (node: Node, parent?: Parent) => {
+    if ('children' in node) {
+      const nodeAsParent = node as Parent;
+      nodeAsParent.children.forEach((child) => {
+        visitor(child, nodeAsParent);
+      });
+    }
+
+    if (node.type === 'code' && node.lang === 'esql') {
+      node.type = 'esql';
+    }
+  };
+
+  return (tree: Node) => {
+    visitor(tree);
+  };
+};
+
+export function MessageText({ loading, content, onRunQueryClick: onRunQueryClickProp }: Props) {
   const containerClassName = css`
     overflow-wrap: break-word;
   `;
 
+  const onRunQueryClickRef = useRef(onRunQueryClickProp);
+
+  onRunQueryClickRef.current = onRunQueryClickProp;
+
   const { parsingPluginList, processingPluginList } = useMemo(() => {
     const parsingPlugins = getDefaultEuiMarkdownParsingPlugins();
+
     const processingPlugins = getDefaultEuiMarkdownProcessingPlugins();
 
     const { components } = processingPlugins[1][1];
 
     processingPlugins[1][1].components = {
```

Comment:
```suggestion
    components = {
```

Reply:
we're intentionally mutating, we're returning processingPlugins again at the end of the function. This would create an entirely new variable which wouldn't be used.